frontend/javascripts/oxalis/view/right-border-tabs/connectome_tab/connectome_view.js

```diff
@@ -0,0 +1,784 @@
+    const checkedNodeKeyPrefix = `segment;${node.data.id};`;
+    if (isChecked) {
+      // Find out which keys should also be checked, because they represent the same agglomerate
+      const treeData = convertConnectomeToTreeData(this.state.connectomeData) || [];
+      const additionalCheckedKeys = Array.from(
+        mapAndFilterTreeData(
+          treeData,
+          treeNode => treeNode.key,
+          treeNode => treeNode.key.startsWith(checkedNodeKeyPrefix),
+        ),
+      );
+      this.setState({ checkedKeys: [...checked, ...additionalCheckedKeys] });
+    } else {
+      // Find out which keys should also be unchecked, because they represent the same agglomerate
+      const checkedKeys = checked.filter(key => !key.startsWith(checkedNodeKeyPrefix));
+      this.setState({ checkedKeys });
+    }
+  };
+
+  handleExpand = (expandedKeys: Array<string>) => {
+    this.setState({ expandedKeys });
+  };
+
+  isConnectomeMappingActive(): boolean {
+    const { mappingInfo, currentConnectomeFile } = this.props;
+
+    if (mappingInfo == null || currentConnectomeFile == null) return true;
+
+    if (
+      mappingInfo.mappingName !== currentConnectomeFile.mappingName ||
+      mappingInfo.mappingStatus === MappingStatusEnum.DISABLED
+    ) {
+      return false;
+    }
+    return true;
+  }
+
+  activateConnectomeMapping() {
+    const { segmentationLayer, currentConnectomeFile } = this.props;
+
+    if (segmentationLayer == null || currentConnectomeFile == null) return;
+
+    Store.dispatch(
+      setMappingAction(segmentationLayer.name, currentConnectomeFile.mappingName, "HDF5", {
+        showLoadingIndicator: true,
+      }),
+    );
+  }
+
+  getConnectomeMappingActivationAlert() {
+    const isConnectomeMappingActive = this.isConnectomeMappingActive();
+
+    return isConnectomeMappingActive ? null : (
+      <Alert
+        message={
+          <>
+            The mapping this connectome was computed for is not active.{" "}
+            <a href="#" onClick={() => this.activateConnectomeMapping()}>
+              Click to activate.
+            </a>
+          </>
+        }
+        type="info"
+        showIcon
+        style={{ marginBottom: 10 }}
+      />
+    );
+  }
+
+  getConnectomeHeader() {
+    const { activeAgglomerateIds, segmentationLayer, currentConnectomeFile } = this.props;
+    const { availableSynapseTypes, connectomeData } = this.state;
+    const activeAgglomerateIdString = activeAgglomerateIds.length
+      ? activeAgglomerateIds.join(",")
+      : "";
+
+    const disabled = currentConnectomeFile == null;
+
+    return (
+      <>
+        <Input.Group compact className="compact-icons" style={{ marginBottom: 10 }}>
+          <Tooltip title="Show Synaptic Connections for Segment ID(s)">
+            <InputComponent
+              value={activeAgglomerateIdString}
+              onPressEnter={this.handleChangeActiveSegment}
+              placeholder="Segment ID 1[, Segment ID 2, ...]"
```

Comment:
I'm not sure whether the placeholder could confuse people so that they try to input "Segment ID 1". Maybe "Enter one or multiple segment ids here (separated by commas)" would be clearer?

Reply:
Didn't think of that, but you're likely right. Since the placeholder would get quite long, I'd try it with "Enter Segment ID(s)" and expect users to figure out (or understand intuitively) that multiple values need to be comma-separated :thinking:

Reply:
Related, I thought about using a multi-select with a "badge-style" instead of the standard input field. This would be configurable to auto-create badges when entering "," or " ". Also, it would be easier to remove single segment IDs and look better imo. See this sandbox for an example: https://codesandbox.io/s/automatic-tokenization-antd-4-18-3-forked-b13ig
Enter a segment ID and press Enter or type "," or " ".

Reply:
Interesting! I agree with the advantages you mention. I only want to bring up one disadvantage: One cannot easily copy all active segment ids, anymore (pasting obviously works). If one wants to make notes in an external text file or something like that, this could be a bit annoying. However, this scenario is a bit far fetched and could easily be fixed with a dedicated "copy segment ids" button. Just wanted to mention it.